# Introduction To Probability
## Challenge 1

A and B are events of a probability space with $(\omega, \sigma, P)$ such that $P(A) = 0.3$, $P(B) = 0.6$ and $P(A \cap B) = 0.1$

Which of the following statements are false?
* $P(A \cup B) = 0.6$
* $P(A \cap B^{C}) = 0.2$
* $P(A \cap (B \cup B^{C})) = 0.4$
* $P(A^{C} \cap B^{C}) = 0.3$
* $P((A \cap B)^{C}) = 0.9$

**ANSWER**
* $P(A \cup B) = P(A) + P(B) - P(A \cap B) = 0.3 + 0.6 - 0.1 = 0.8$                **FALSE**
* $P(A \cap B^{C}) = P(A) - P(A \cap B) = 0.3 - 0.1 = 0.2$                         **TRUE**
* $P(A \cap (B \cup B^{C})) = P (A \cap 1) = P(A) = 0.3$                           **FALSE**
* $P(A^{C} \cap B^{C}) = P(B^{C}) - P(A \cap B^{C}) = (1 - 0.6) - 0.2 = 0.2$       **FALSE**
* $P((A \cap B)^{C}) = 1 - P(A \cap B) = 1 - 0.1 = 0.9$                            **TRUE**

## Challenge 2
There is a box with 10 white balls, 12 red balls and 8 black balls. Calculate the probability of:
* Taking a white ball out.
* Taking a white ball out after taking a black ball out.
* Taking a red ball out after taking a black and a red ball out.
* Taking a red ball out after taking a black and a red ball out with reposition.

**Hint**: Reposition means putting back the ball into the box after taking it out.

**ANSWER**
* $P(W) = 10/30 = 1/3 = 0.33$
* $P(W \cap B) = P(B) * P(W | B) = 8/30 * 10/29 = 0.09195$
* $P(R \cap (B \cap R)) = P(R) * P(B | R) * P(R | (B \cap R)) = 8/30 * 12/29 * 11/28 = 0.04335$
* $P(R \cap (B \cap R)) = = 8/30 * 12/30 * 12/30 = 0.04267$

## Challenge 3

You are planning to go on a picnic today but the morning is cloudy. You hate rain so you don't know whether to go out or stay home! To help you make a decision, you gather the following data about rainy days:

* 50% of all rainy days start off cloudy!
* Cloudy mornings are common. About 40% of days start cloudy. 
* This month is usually dry so only 3 of 30 days (10%) tend to be rainy. 

What is the chance of rain during the day?

**ANSWER** <br>
**Information:** <br>
$P(Cloudy | Rainy) = 0.5$ <br>
$P(Cloudy) = 0.4$ <br>
$P(Rainy) = 0.1$ <br>

**Calculation ( Bayes Theorem)** <br>
$P(Rainy | Cloudy) = P(Cloudy | Rainy) * P(Rainy) / P(Cloudy) = 0.5 * 0.1 / 0.4 = 0.125$

## Challenge 4

One thousand people were asked through a telephone survey whether they thought more street lighting is needed at night or not.

Out of the 480 men that answered the survey, 324 said yes and 156 said no. On the other hand, out of the 520 women that answered, 351 said yes and 169 said no. 

We wonder if men and women have a different opinions about the street lighting matter. Is gender relevant or irrelevant to the question?

Consider the following events:
- The answer is yes, so the person that answered thinks that more street lighting is needed.
- The person who answered is a man.

We want to know if these events are independent, that is, if the fact of wanting more light depends on whether one is male or female. Are these events independent or not?

**Hint**: To clearly compare the answers by gender, it is best to place the data in a table.

In [6]:
import pandas as pd
table = pd.DataFrame([{'Yes': 0.324, 'No':0.156, 'Total' : 0.480}, 
                      {'Yes' : 0.351, 'No' : 0.169, 'Total' : 0.520}, 
                      {'Yes':(0.351 + 0.324), 'No' : (0.156+0.169), 'Total' : 1}], 
                     index = ['Men','Women', 'Total'])
table

Yes     No  Total
Men    0.324  0.156   0.48
Women  0.351  0.169   0.52
Total  0.675  0.325   1.00

$P(Yes) = 0.675$ 
$P(Men) = 0.48$ 

$P(Yes)*P(Men) = 0.675 * 0.48 = 0.324 = P(Yes \cap Men)$

**These variables are independent**